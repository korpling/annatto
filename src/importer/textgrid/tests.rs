use std::io::BufWriter;

use super::*;
use graphannis::AnnotationGraph;
use pretty_assertions::assert_eq;

#[test]
fn parse_tier_groups_param() {
    let result = parse_tier_map("A={lemma,pos,Inf-Struct};B={}");
    assert_eq!(2, result.len());
    let a = result.get("A").unwrap();
    assert_eq!(3, a.len());
    assert!(a.contains("lemma"));
    assert!(a.contains("pos"));
    assert!(a.contains("Inf-Struct"));
    let b = result.get("B").unwrap();
    assert_eq!(0, b.len());
}

#[test]
fn single_speaker() {
    let mut properties: BTreeMap<String, String> = BTreeMap::new();
    properties.insert(
        "tier_groups".to_string(),
        "tok={lemma,pos,Inf-Struct}".to_string(),
    );
    properties.insert("skip_timeline_generation".to_string(), "true".to_string());
    let importer = TextgridImporter::default();

    let mut u = importer
        .import_corpus(
            &PathBuf::from("tests/data/textgrid/in/singleSpeaker"),
            &properties,
            None,
        )
        .unwrap();
    let mut g = AnnotationGraph::with_default_graphstorages(false).unwrap();
    g.apply_update(&mut u, |_| {}).unwrap();

    let mut buf = BufWriter::new(Vec::new());
    graphannis_core::graph::serialization::graphml::export(&g, None, &mut buf, |_| {}).unwrap();
    let bytes = buf.into_inner().unwrap();
    let actual = String::from_utf8(bytes).unwrap();

    // Compare the actual output with the expected one
    assert_eq!(
        include_str!("../../../tests/data/textgrid/out/singleSpeaker.graphml"),
        actual
    );
}

#[test]
fn two_speakers() {
    let mut properties: BTreeMap<String, String> = BTreeMap::new();
    properties.insert(
        "tier_groups".to_string(),
        "A={lemma,pos,Inf-Struct};B={}".to_string(),
    );
    let importer = TextgridImporter::default();

    let mut u = importer
        .import_corpus(
            &PathBuf::from("tests/data/textgrid/in/twoSpeakers"),
            &properties,
            None,
        )
        .unwrap();
    let mut g = AnnotationGraph::with_default_graphstorages(false).unwrap();
    g.apply_update(&mut u, |_| {}).unwrap();

    let mut buf = BufWriter::new(Vec::new());
    graphannis_core::graph::serialization::graphml::export(&g, None, &mut buf, |_| {}).unwrap();
    let bytes = buf.into_inner().unwrap();
    let actual = String::from_utf8(bytes).unwrap();

    // Compare the actual output with the expected one
    assert_eq!(
        include_str!("../../../tests/data/textgrid/out/twoSpeakers.graphml"),
        actual
    );
}

#[test]
fn misaligned_lemma_annotation() {
    let mut properties: BTreeMap<String, String> = BTreeMap::new();
    properties.insert(
        "tier_groups".to_string(),
        "A={lemma,pos,Inf-Struct};B={}".to_string(),
    );
    let importer = TextgridImporter::default();

    let mut u = importer
        .import_corpus(
            &PathBuf::from("tests/data/textgrid/in/misalignedLemma"),
            &properties,
            None,
        )
        .unwrap();
    let mut g = AnnotationGraph::with_default_graphstorages(false).unwrap();
    g.apply_update(&mut u, |_| {}).unwrap();

    let mut buf = BufWriter::new(Vec::new());
    graphannis_core::graph::serialization::graphml::export(&g, None, &mut buf, |_| {}).unwrap();
    let bytes = buf.into_inner().unwrap();
    let actual = String::from_utf8(bytes).unwrap();

    // Compare the actual output with the expected one
    assert_eq!(
        include_str!("../../../tests/data/textgrid/out/misalignedLemma.graphml").trim(),
        actual.trim()
    );
}

#[test]
fn needs_tier_groups_property() {
    let properties: BTreeMap<String, String> = BTreeMap::new();

    let importer = TextgridImporter::default();
    let result = importer.import_corpus(
        &PathBuf::from("tests/data/textgrid/in/singleSpeaker"),
        &properties,
        None,
    );
    assert_eq!(true, result.is_err());
}
