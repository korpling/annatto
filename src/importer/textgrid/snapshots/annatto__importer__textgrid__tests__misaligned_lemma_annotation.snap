---
source: src/importer/textgrid/tests.rs
expression: actual
---
<?xml version="1.0" encoding="UTF-8"?>
<graphml>
    <key id="k0" for="node" attr.name="A" attr.type="string"/>
    <key id="k1" for="node" attr.name="B" attr.type="string"/>
    <key id="k2" for="node" attr.name="Inf-Struct" attr.type="string"/>
    <key id="k3" for="node" attr.name="annis::doc" attr.type="string"/>
    <key id="k4" for="node" attr.name="annis::layer" attr.type="string"/>
    <key id="k5" for="node" attr.name="lemma" attr.type="string"/>
    <key id="k6" for="node" attr.name="annis::node_type" attr.type="string"/>
    <key id="k7" for="node" attr.name="pos" attr.type="string"/>
    <key id="k8" for="node" attr.name="annis::time" attr.type="string"/>
    <key id="k9" for="node" attr.name="annis::tok" attr.type="string"/>
    <key id="k10" for="node" attr.name="annis::tok-whitespace-after" attr.type="string"/>
    <graph edgedefault="directed" parse.order="nodesfirst" parse.nodeids="free" parse.edgeids="canonical">
        <node id="misalignedLemma">
            <data key="k6">corpus</data>
        </node>
        <node id="misalignedLemma/doc1">
            <data key="k3">doc1</data>
            <data key="k6">corpus</data>
        </node>
        <node id="misalignedLemma/doc1#text">
            <data key="k6">datasource</data>
        </node>
        <node id="misalignedLemma/doc1#t1">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">0-1</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t2">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">1-2</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t3">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">2-3</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t4">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">3-4</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t5">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">4-5</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t6">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">5-6</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t7">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">6-7</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t8">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">7-8</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t9">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">8-9</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t10">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">9-10</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t11">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">10-11</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t12">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">11-12</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t13">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">12-13</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#t14">
            <data key="k10"> </data>
            <data key="k6">node</data>
            <data key="k8">13-</data>
            <data key="k9"></data>
        </node>
        <node id="misalignedLemma/doc1#sSpan1">
            <data key="k0">Is</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">Is</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan2">
            <data key="k0">this</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">this</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan3">
            <data key="k0">example</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">example</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan4">
            <data key="k0">more</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">more</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan5">
            <data key="k0">complicated</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">complicated</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan6">
            <data key="k0">than</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">than</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan7">
            <data key="k0">it</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">it</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan8">
            <data key="k0">appears</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">appears</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan9">
            <data key="k0">to</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">to</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan10">
            <data key="k0">be</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">be</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan11">
            <data key="k0">?</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">?</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan12">
            <data key="k2">contrast-focus</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan13">
            <data key="k2">topic</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan14">
            <data key="k4">default_ns</data>
            <data key="k5">be</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan15">
            <data key="k4">default_ns</data>
            <data key="k5">this</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan16">
            <data key="k4">default_ns</data>
            <data key="k5">example</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan17">
            <data key="k4">default_ns</data>
            <data key="k5">more</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan18">
            <data key="k4">default_ns</data>
            <data key="k5">complicated</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan19">
            <data key="k4">default_ns</data>
            <data key="k5">than</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan20">
            <data key="k4">default_ns</data>
            <data key="k5">it</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan21">
            <data key="k4">default_ns</data>
            <data key="k5">appear</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan22">
            <data key="k4">default_ns</data>
            <data key="k5">to</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan23">
            <data key="k4">default_ns</data>
            <data key="k5">be</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan24">
            <data key="k4">default_ns</data>
            <data key="k5">?</data>
            <data key="k6">node</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan25">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">VBZ</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan26">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">DT</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan27">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">NN</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan28">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">RBR</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan29">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">JJ</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan30">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">IN</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan31">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">PRP</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan32">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">VBZ</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan33">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">TO</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan34">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">VB</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan35">
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k7">.</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan36">
            <data key="k1">um</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">um</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan37">
            <data key="k1">No</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">No</data>
        </node>
        <node id="misalignedLemma/doc1#sSpan38">
            <data key="k1">!</data>
            <data key="k4">default_ns</data>
            <data key="k6">node</data>
            <data key="k9">!</data>
        </node>
        <edge id="e0" source="misalignedLemma/doc1#sSpan36" target="misalignedLemma/doc1#t5" label="Coverage/annis/">
        </edge>
        <edge id="e1" source="misalignedLemma/doc1#sSpan36" target="misalignedLemma/doc1#t6" label="Coverage/annis/">
        </edge>
        <edge id="e2" source="misalignedLemma/doc1#sSpan36" target="misalignedLemma/doc1#t7" label="Coverage/annis/">
        </edge>
        <edge id="e3" source="misalignedLemma/doc1#sSpan33" target="misalignedLemma/doc1#t9" label="Coverage/annis/">
        </edge>
        <edge id="e4" source="misalignedLemma/doc1#sSpan30" target="misalignedLemma/doc1#t6" label="Coverage/annis/">
        </edge>
        <edge id="e5" source="misalignedLemma/doc1#sSpan27" target="misalignedLemma/doc1#t3" label="Coverage/annis/">
        </edge>
        <edge id="e6" source="misalignedLemma/doc1#sSpan24" target="misalignedLemma/doc1#t11" label="Coverage/annis/">
        </edge>
        <edge id="e7" source="misalignedLemma/doc1#sSpan21" target="misalignedLemma/doc1#t8" label="Coverage/annis/">
        </edge>
        <edge id="e8" source="misalignedLemma/doc1#sSpan18" target="misalignedLemma/doc1#t5" label="Coverage/annis/">
        </edge>
        <edge id="e9" source="misalignedLemma/doc1#sSpan15" target="misalignedLemma/doc1#t2" label="Coverage/annis/">
        </edge>
        <edge id="e10" source="misalignedLemma/doc1#sSpan12" target="misalignedLemma/doc1#t1" label="Coverage/annis/">
        </edge>
        <edge id="e11" source="misalignedLemma/doc1#sSpan9" target="misalignedLemma/doc1#t9" label="Coverage/annis/">
        </edge>
        <edge id="e12" source="misalignedLemma/doc1#sSpan6" target="misalignedLemma/doc1#t6" label="Coverage/annis/">
        </edge>
        <edge id="e13" source="misalignedLemma/doc1#sSpan3" target="misalignedLemma/doc1#t3" label="Coverage/annis/">
        </edge>
        <edge id="e14" source="misalignedLemma/doc1#sSpan37" target="misalignedLemma/doc1#t12" label="Coverage/annis/">
        </edge>
        <edge id="e15" source="misalignedLemma/doc1#sSpan34" target="misalignedLemma/doc1#t10" label="Coverage/annis/">
        </edge>
        <edge id="e16" source="misalignedLemma/doc1#sSpan31" target="misalignedLemma/doc1#t7" label="Coverage/annis/">
        </edge>
        <edge id="e17" source="misalignedLemma/doc1#sSpan28" target="misalignedLemma/doc1#t4" label="Coverage/annis/">
        </edge>
        <edge id="e18" source="misalignedLemma/doc1#sSpan25" target="misalignedLemma/doc1#t1" label="Coverage/annis/">
        </edge>
        <edge id="e19" source="misalignedLemma/doc1#sSpan22" target="misalignedLemma/doc1#t9" label="Coverage/annis/">
        </edge>
        <edge id="e20" source="misalignedLemma/doc1#sSpan19" target="misalignedLemma/doc1#t6" label="Coverage/annis/">
        </edge>
        <edge id="e21" source="misalignedLemma/doc1#sSpan16" target="misalignedLemma/doc1#t3" label="Coverage/annis/">
        </edge>
        <edge id="e22" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t2" label="Coverage/annis/">
        </edge>
        <edge id="e23" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t3" label="Coverage/annis/">
        </edge>
        <edge id="e24" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t4" label="Coverage/annis/">
        </edge>
        <edge id="e25" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t5" label="Coverage/annis/">
        </edge>
        <edge id="e26" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t6" label="Coverage/annis/">
        </edge>
        <edge id="e27" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t7" label="Coverage/annis/">
        </edge>
        <edge id="e28" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t8" label="Coverage/annis/">
        </edge>
        <edge id="e29" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t9" label="Coverage/annis/">
        </edge>
        <edge id="e30" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t10" label="Coverage/annis/">
        </edge>
        <edge id="e31" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#t11" label="Coverage/annis/">
        </edge>
        <edge id="e32" source="misalignedLemma/doc1#sSpan10" target="misalignedLemma/doc1#t10" label="Coverage/annis/">
        </edge>
        <edge id="e33" source="misalignedLemma/doc1#sSpan7" target="misalignedLemma/doc1#t7" label="Coverage/annis/">
        </edge>
        <edge id="e34" source="misalignedLemma/doc1#sSpan4" target="misalignedLemma/doc1#t4" label="Coverage/annis/">
        </edge>
        <edge id="e35" source="misalignedLemma/doc1#sSpan1" target="misalignedLemma/doc1#t1" label="Coverage/annis/">
        </edge>
        <edge id="e36" source="misalignedLemma/doc1#sSpan38" target="misalignedLemma/doc1#t13" label="Coverage/annis/">
        </edge>
        <edge id="e37" source="misalignedLemma/doc1#sSpan35" target="misalignedLemma/doc1#t11" label="Coverage/annis/">
        </edge>
        <edge id="e38" source="misalignedLemma/doc1#sSpan32" target="misalignedLemma/doc1#t8" label="Coverage/annis/">
        </edge>
        <edge id="e39" source="misalignedLemma/doc1#sSpan29" target="misalignedLemma/doc1#t5" label="Coverage/annis/">
        </edge>
        <edge id="e40" source="misalignedLemma/doc1#sSpan26" target="misalignedLemma/doc1#t2" label="Coverage/annis/">
        </edge>
        <edge id="e41" source="misalignedLemma/doc1#sSpan23" target="misalignedLemma/doc1#t10" label="Coverage/annis/">
        </edge>
        <edge id="e42" source="misalignedLemma/doc1#sSpan20" target="misalignedLemma/doc1#t7" label="Coverage/annis/">
        </edge>
        <edge id="e43" source="misalignedLemma/doc1#sSpan17" target="misalignedLemma/doc1#t4" label="Coverage/annis/">
        </edge>
        <edge id="e44" source="misalignedLemma/doc1#sSpan14" target="misalignedLemma/doc1#t1" label="Coverage/annis/">
        </edge>
        <edge id="e45" source="misalignedLemma/doc1#sSpan11" target="misalignedLemma/doc1#t11" label="Coverage/annis/">
        </edge>
        <edge id="e46" source="misalignedLemma/doc1#sSpan8" target="misalignedLemma/doc1#t8" label="Coverage/annis/">
        </edge>
        <edge id="e47" source="misalignedLemma/doc1#sSpan5" target="misalignedLemma/doc1#t5" label="Coverage/annis/">
        </edge>
        <edge id="e48" source="misalignedLemma/doc1#sSpan2" target="misalignedLemma/doc1#t2" label="Coverage/annis/">
        </edge>
        <edge id="e49" source="misalignedLemma/doc1#t13" target="misalignedLemma/doc1#t14" label="Ordering/annis/">
        </edge>
        <edge id="e50" source="misalignedLemma/doc1#t12" target="misalignedLemma/doc1#t13" label="Ordering/annis/">
        </edge>
        <edge id="e51" source="misalignedLemma/doc1#t11" target="misalignedLemma/doc1#t12" label="Ordering/annis/">
        </edge>
        <edge id="e52" source="misalignedLemma/doc1#t10" target="misalignedLemma/doc1#t11" label="Ordering/annis/">
        </edge>
        <edge id="e53" source="misalignedLemma/doc1#t9" target="misalignedLemma/doc1#t10" label="Ordering/annis/">
        </edge>
        <edge id="e54" source="misalignedLemma/doc1#t8" target="misalignedLemma/doc1#t9" label="Ordering/annis/">
        </edge>
        <edge id="e55" source="misalignedLemma/doc1#t7" target="misalignedLemma/doc1#t8" label="Ordering/annis/">
        </edge>
        <edge id="e56" source="misalignedLemma/doc1#t6" target="misalignedLemma/doc1#t7" label="Ordering/annis/">
        </edge>
        <edge id="e57" source="misalignedLemma/doc1#t5" target="misalignedLemma/doc1#t6" label="Ordering/annis/">
        </edge>
        <edge id="e58" source="misalignedLemma/doc1#t4" target="misalignedLemma/doc1#t5" label="Ordering/annis/">
        </edge>
        <edge id="e59" source="misalignedLemma/doc1#t3" target="misalignedLemma/doc1#t4" label="Ordering/annis/">
        </edge>
        <edge id="e60" source="misalignedLemma/doc1#t2" target="misalignedLemma/doc1#t3" label="Ordering/annis/">
        </edge>
        <edge id="e61" source="misalignedLemma/doc1#t1" target="misalignedLemma/doc1#t2" label="Ordering/annis/">
        </edge>
        <edge id="e62" source="misalignedLemma/doc1#sSpan10" target="misalignedLemma/doc1#sSpan11" label="Ordering/default_ns/A">
        </edge>
        <edge id="e63" source="misalignedLemma/doc1#sSpan7" target="misalignedLemma/doc1#sSpan8" label="Ordering/default_ns/A">
        </edge>
        <edge id="e64" source="misalignedLemma/doc1#sSpan4" target="misalignedLemma/doc1#sSpan5" label="Ordering/default_ns/A">
        </edge>
        <edge id="e65" source="misalignedLemma/doc1#sSpan1" target="misalignedLemma/doc1#sSpan2" label="Ordering/default_ns/A">
        </edge>
        <edge id="e66" source="misalignedLemma/doc1#sSpan8" target="misalignedLemma/doc1#sSpan9" label="Ordering/default_ns/A">
        </edge>
        <edge id="e67" source="misalignedLemma/doc1#sSpan5" target="misalignedLemma/doc1#sSpan6" label="Ordering/default_ns/A">
        </edge>
        <edge id="e68" source="misalignedLemma/doc1#sSpan2" target="misalignedLemma/doc1#sSpan3" label="Ordering/default_ns/A">
        </edge>
        <edge id="e69" source="misalignedLemma/doc1#sSpan9" target="misalignedLemma/doc1#sSpan10" label="Ordering/default_ns/A">
        </edge>
        <edge id="e70" source="misalignedLemma/doc1#sSpan6" target="misalignedLemma/doc1#sSpan7" label="Ordering/default_ns/A">
        </edge>
        <edge id="e71" source="misalignedLemma/doc1#sSpan3" target="misalignedLemma/doc1#sSpan4" label="Ordering/default_ns/A">
        </edge>
        <edge id="e72" source="misalignedLemma/doc1#sSpan36" target="misalignedLemma/doc1#sSpan37" label="Ordering/default_ns/B">
        </edge>
        <edge id="e73" source="misalignedLemma/doc1#sSpan37" target="misalignedLemma/doc1#sSpan38" label="Ordering/default_ns/B">
        </edge>
        <edge id="e74" source="misalignedLemma/doc1#sSpan36" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e75" source="misalignedLemma/doc1#sSpan33" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e76" source="misalignedLemma/doc1#sSpan30" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e77" source="misalignedLemma/doc1#sSpan27" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e78" source="misalignedLemma/doc1#sSpan24" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e79" source="misalignedLemma/doc1#sSpan21" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e80" source="misalignedLemma/doc1#sSpan18" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e81" source="misalignedLemma/doc1#sSpan15" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e82" source="misalignedLemma/doc1#sSpan12" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e83" source="misalignedLemma/doc1#sSpan9" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e84" source="misalignedLemma/doc1#sSpan6" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e85" source="misalignedLemma/doc1#sSpan3" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e86" source="misalignedLemma/doc1#t14" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e87" source="misalignedLemma/doc1#t11" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e88" source="misalignedLemma/doc1#t8" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e89" source="misalignedLemma/doc1#t5" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e90" source="misalignedLemma/doc1#t2" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e91" source="misalignedLemma/doc1" target="misalignedLemma" label="PartOf/annis/">
        </edge>
        <edge id="e92" source="misalignedLemma/doc1#sSpan37" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e93" source="misalignedLemma/doc1#sSpan34" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e94" source="misalignedLemma/doc1#sSpan31" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e95" source="misalignedLemma/doc1#sSpan28" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e96" source="misalignedLemma/doc1#sSpan25" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e97" source="misalignedLemma/doc1#sSpan22" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e98" source="misalignedLemma/doc1#sSpan19" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e99" source="misalignedLemma/doc1#sSpan16" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e100" source="misalignedLemma/doc1#sSpan13" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e101" source="misalignedLemma/doc1#sSpan10" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e102" source="misalignedLemma/doc1#sSpan7" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e103" source="misalignedLemma/doc1#sSpan4" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e104" source="misalignedLemma/doc1#sSpan1" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e105" source="misalignedLemma/doc1#t12" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e106" source="misalignedLemma/doc1#t9" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e107" source="misalignedLemma/doc1#t6" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e108" source="misalignedLemma/doc1#t3" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e109" source="misalignedLemma/doc1#text" target="misalignedLemma/doc1" label="PartOf/annis/">
        </edge>
        <edge id="e110" source="misalignedLemma/doc1#sSpan38" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e111" source="misalignedLemma/doc1#sSpan35" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e112" source="misalignedLemma/doc1#sSpan32" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e113" source="misalignedLemma/doc1#sSpan29" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e114" source="misalignedLemma/doc1#sSpan26" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e115" source="misalignedLemma/doc1#sSpan23" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e116" source="misalignedLemma/doc1#sSpan20" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e117" source="misalignedLemma/doc1#sSpan17" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e118" source="misalignedLemma/doc1#sSpan14" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e119" source="misalignedLemma/doc1#sSpan11" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e120" source="misalignedLemma/doc1#sSpan8" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e121" source="misalignedLemma/doc1#sSpan5" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e122" source="misalignedLemma/doc1#sSpan2" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e123" source="misalignedLemma/doc1#t13" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e124" source="misalignedLemma/doc1#t10" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e125" source="misalignedLemma/doc1#t7" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e126" source="misalignedLemma/doc1#t4" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
        <edge id="e127" source="misalignedLemma/doc1#t1" target="misalignedLemma/doc1#text" label="PartOf/annis/">
        </edge>
    </graph>
</graphml>
